slack_sdk/models/blocks/block_elements.py

```diff
@@ -64,64 +64,31 @@ def parse(cls, block_element: Union[dict, "BlockElement"]) -> Optional[Union["Bl
-                elif t == ImageElement.type:
-                    return ImageElement(**d)
-                elif t == ButtonElement.type:
-                    return ButtonElement(**d)
-                elif t == StaticSelectElement.type:
-                    return StaticSelectElement(**d)
-                elif t == StaticMultiSelectElement.type:
-                    return StaticMultiSelectElement(**d)
-                elif t == ExternalDataSelectElement.type:
-                    return ExternalDataSelectElement(**d)
-                elif t == ExternalDataMultiSelectElement.type:
-                    return ExternalDataMultiSelectElement(**d)
-                elif t == UserSelectElement.type:
-                    return UserSelectElement(**d)
-                elif t == UserMultiSelectElement.type:
-                    return UserMultiSelectElement(**d)
-                elif t == ConversationSelectElement.type:
-                    return ConversationSelectElement(**d)
-                elif t == ConversationMultiSelectElement.type:
-                    return ConversationMultiSelectElement(**d)
-                elif t == ChannelSelectElement.type:
-                    return ChannelSelectElement(**d)
-                elif t == ChannelMultiSelectElement.type:
-                    return ChannelMultiSelectElement(**d)
-                elif t == PlainTextInputElement.type:
-                    return PlainTextInputElement(**d)
-                elif t == RadioButtonsElement.type:
-                    return RadioButtonsElement(**d)
-                elif t == CheckboxesElement.type:
-                    return CheckboxesElement(**d)
-                elif t == OverflowMenuElement.type:
-                    return OverflowMenuElement(**d)
-                elif t == DatePickerElement.type:
-                    return DatePickerElement(**d)
-                elif t == TimePickerElement.type:
-                    return TimePickerElement(**d)
-                else:
-                    cls.logger.warning(f"Unknown element detected and skipped ({block_element})")
-                    return None
-            else:
-                cls.logger.warning(f"Unknown element detected and skipped ({block_element})")
-                return None
         elif isinstance(block_element, (TextObject, BlockElement)):
             return block_element
-        else:
-            cls.logger.warning(f"Unknown element detected and skipped ({block_element})")
-            return None
+        cls.logger.warning(f"Unknown element detected and skipped ({block_element})")
+        return None
 
     @classmethod
     def parse_all(
         cls, block_elements: Sequence[Union[dict, "BlockElement", TextObject]]
     ) -> List[Union["BlockElement", TextObject]]:
         return [cls.parse(e) for e in block_elements or []]
 
+    @classmethod
+    def _get_sub_block_elements(cls: Type["BlockElement"]) -> Iterator[Type["BlockElement"]]:
```

Comment:
Nice refactoring. If we observe slower runtime performance, we may consider revert this (but probably no)

Reply:
I was also concerned about the runtime performance of this, so I ran some tests,

- best case is about the same as before since this is using a generator
- worst case when all the subclasses must be found, is less then 0.1 ms slower on my local

But I agree if this does not scale properly and causes issues we can remove it